yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/internal/bytecode/types/gambit/ExpressionHandler.java

```diff
@@ -893,6 +914,17 @@ public RecordBuilder add(Location loc, String fieldName, BytecodeExpression inpu
                 dynamicBuilder.add(loc, fieldName, input);
                 return this;
             }
+            if (!isJavaFieldName(fieldName)) {
+              if (null == dynamicBuilder) {
```

Comment:
dynamicBuilder will never be null when dynamic is true and this will never be reached unless dynamic is not yet true (since the first few lines of the method check for that).

Maybe extract the other "switch to dynamic" block into a private method and then call that here? Lines 945-951 -- https://github.com/dsyzhu/yql-plus/blob/cbb88489e6f367a09b95883a48c27fa0b708474d/yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/internal/bytecode/types/gambit/ExpressionHandler.java#L945-L951